test/micromamba/test_shell.py

```diff
@@ -152,20 +158,22 @@ def custom_shell(shell):
             assert res["actions"]["print"][0] == ""
 
     @pytest.mark.parametrize("shell_type", ["bash", "posix", "powershell", "cmd.exe"])
-    @pytest.mark.parametrize("root", [False, True])
-    @pytest.mark.parametrize("env_exists", [False, True])
+    @pytest.mark.parametrize("prefix_is_root", [False, True])
+    @pytest.mark.parametrize("prefix_exists", [False, True])
     @pytest.mark.parametrize("expanded_home", [False, True])
     @pytest.mark.parametrize("prefix_type", ["prefix", "name"])
-    def test_activate(self, shell_type, root, env_exists, prefix_type, expanded_home):
+    def test_activate(
+        self, shell_type, prefix_is_root, prefix_exists, prefix_type, expanded_home
+    ):
         skip_if_shell_incompat(shell_type)
 
-        if env_exists:
+        if prefix_exists:
             # Create the environment for this test, so that it exists
             create("-n", TestShell.env_name, "-q", "--offline", no_dry_run=True)
         else:
             shutil.rmtree(TestShell.root_prefix)
```

Comment:
This may not cover `prefix_is_root = False`. 
```suggestion
            if prefix_is_root:
                shutil.rmtree(TestShell.root_prefix)
            else:
                shutil.rmtree(TestShell.prefix)
```

Reply:
I don't get this one, the target prefix of the test is located at root_prefix/envs/env_name == target_prefix.
So removing the root_prefix that is always created at test setup as an empty dir should do the job right ?

Reply:
But yeah it's more *accurate* to consider that only the target_prefix doesn't exist. And when it's not the root prefix, we keep the root prefix.
The else condition is not necessary since at test setup we always get an empty root prefix
Will do that.